store/remote.go

```diff
@@ -0,0 +1,53 @@
+package store
+
+import (
+	"time"
+
+	"github.com/gluster/glusterd2/pkg/elasticetcd"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/clientv3/concurrency"
+)
+
+func newRemoteStore(conf *Config) (*GDStore, error) {
+	if len(conf.Endpoints) == 1 && conf.Endpoints[1] == elasticetcd.DefaultEndpoint {
+		return nil, ErrEndpointsRequired
+	}
+
+	c, e := clientv3.New(clientv3.Config{
+		Endpoints:        conf.Endpoints,
+		AutoSyncInterval: 1 * time.Minute,
+		DialTimeout:      10 * time.Second,
+	})
+	if e != nil {
+		log.WithError(e).Error("failed to create etcd client")
+		return nil, e
+	}
+	log.Debug("etcd client connection created")
+
+	// Create a new session (lease kept alive for the lifetime of a client)
+	// This is currently used for:
+	// * distributed locking (Mutex)
+	// * representing liveness of the client
+	s, e := concurrency.NewSession(c, concurrency.WithTTL(sessionTTL))
```

Comment:
You can move the `sessionTTL` to this file as it's the only place where it's used currently. But, it also means that in case of embedded store, when `publishLiveness()` is called, the TTL is different (default for the session i.e 60s)